airbrakes/data_handling/apogee_predictor.py

```diff
@@ -97,87 +113,110 @@ def update(self, processed_data_packets: deque[ProcessedDataPacket]) -> None:
         # the apogee prediction process.
         self._prediction_queue.put(processed_data_packets.copy())
 
-    def _curve_fit_function(self, t, a, b):
-        """
-        This function is only used internally by scipy curve_fit function
-        Defines the function that the curve fit will use
-        """
-        return a * (1 - b * t) ** 4
-
     def _curve_fit(self) -> npt.NDArray[np.float64]:
         """
         Calculates the curve fit function of rotated compensated acceleration
         Uses the function y = A(1-Bt)^4, where A and B are parameters being fit
-
         :return: numpy array with values of A and B
         """
+        # if there is zero or one data points, we can't curve fit. Returning none is handled
+        # in _create_prediction_lookup_table()
+        if any([len(self._cumulative_time_differences) <=2, len(self._accelerations)<=2]):
+            return None
         # curve fit that returns popt: list of fitted parameters, and pcov: list of uncertainties
-        popt = curve_fit(
+        popt, pcov = curve_fit(
             self._curve_fit_function,
             self._cumulative_time_differences,
             self._accelerations,
             p0=CURVE_FIT_INITIAL,
             maxfev=2000,
-        )[0]
+        )
+        uncertainties = np.sqrt(np.diag(pcov))
+        # determines the minimum amount of data points before we can declare apogee converged or not
+        min_length = np.ceil(MIN_PREDICTION_TIME * APOGEE_PREDICTION_FREQUENCY)
```

Comment:
this is so we don't declare our apogee as converged before we reach a time (0.5 sec)

Reply:
Yeah but why, isn't the standard deviation of the errors (`uncertainites`) enough already? Is there a particular case / scenario where you think that would fail?

In any case, this should be a class/module level constant, to avoid recomputation

Reply:
ah yeah you're right. I reverted that change and fixed the corresponding tests.